examples/kalman_filter.ipynb

```diff
@@ -0,0 +1,490 @@
+    "- $y(t)$ denotes the time-dependent output / measurement to the system\n",
+    "- $x(t)$ denotes the time-dependent state of the system (which is not directly measureable in general)\n",
+    "- $f,g$ denote the time-dependent dynamics- and measurement-function, respectively\n",
+    "- $\\epsilon$ denotes random measurement uncertainty\n",
+    "\n",
+    "*The goal is to infer $x(t)$ from $y(1:t)$ even though $f,g,\\epsilon$ are unkown.*\n",
+    "\n",
+    "*A Kalman-Filter represents a possible solution.*\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "!!! info\n",
+    "\n",
+    "    From [Wikipedia](https://en.wikipedia.org/wiki/Kalman_filter): \n",
+    "    \n",
+    "    Kalman filtering, also known as linear quadratic estimation (LQE), is an algorithm that uses a series of measurements observed over time, including statistical noise and other inaccuracies, and produces estimates of unknown variables that tend to be more accurate than those based on a single measurement alone, by estimating a joint probability distribution over the variables for each timeframe. The filter is named after Rudolf E. Kálmán, who was one of the primary developers of its theory.\n",
+    "\n",
+    "    The algorithm works by a two-phase process. For the prediction phase, the Kalman filter produces estimates of the current state variables, along with their uncertainties. Once the outcome of the next measurement (necessarily corrupted with some error, including random noise) is observed, these estimates are updated using a weighted average, with more weight being given to estimates with greater certainty. The algorithm is recursive. It can operate in real time, using only the present input measurements and the state calculated previously and its uncertainty matrix; no additional past information is required.\n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For the sake of simplicity, here we assume that the dynamical system takes the form\n",
+    "\n",
+    "$\\frac{dx}{dt} = Ax(t) + Bu(t)$\n",
+    "\n",
+    "$y(t) = Cx(t) + \\epsilon$ \n",
+    "\n",
+    "where $A,B,C$ are constant matrices, i.e. the system is linear in its state $x(t)$ and its input $u(t)$. Further, the dynamics- and measurement-functions do not depend on time.\n",
+    "\n",
+    "Hence, the above represents the general form of (physical) *linear, time-invariant systems* (LTI systems)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here we define a container object for LTI systems."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class LTISystem(eqx.Module):\n",
+    "    A: jnp.ndarray\n",
+    "    B: jnp.ndarray\n",
+    "    C: jnp.ndarray\n",
+    "\n",
+    "    def __iter__(self):\n",
+    "        return (x for x in (self.A, self.B, self.C))"
```

Comment:
Can we avoid defining this `__iter__` method? It's neat, but it's only used once below.

(Also generator comprehensions are probably worth avoiding in introductory examples; someone will think it's a tuple comprehension.)

Reply:
Sure